dev-tools/mage/manifest/manifest.go

```diff
@@ -0,0 +1,157 @@
+	}
+	return []string{val.URL, val.ShaURL, val.AscURL}
+
+}
+
+// DownloadComponentsFromManifest is going to download a set of components from the given manifest into the destination
+// dropPath folder in order to later use that folder for packaging
+func DownloadComponentsFromManifest(manifest string, platforms []string, platformPackages map[string]string, dropPath string) error {
+	componentSpec := map[string][]string{
+		"apm-server":            {"apm-server"},
+		"beats":                 {"auditbeat", "filebeat", "heartbeat", "metricbeat", "osquerybeat", "packetbeat"},
+		"cloud-defend":          {"cloud-defend"},
+		"cloudbeat":             {"cloudbeat"},
+		"elastic-agent-shipper": {"elastic-agent-shipper"},
+		"endpoint-dev":          {"endpoint-security"},
+		"fleet-server":          {"fleet-server"},
+		"prodfiler":             {"pf-elastic-collector", "pf-elastic-symbolizer", "pf-host-agent"},
+	}
+
+	manifestResponse, err := DownloadManifest(manifest)
+	if err != nil {
+		return fmt.Errorf("failed to download remote manifest file %w", err)
+	}
+	projects := manifestResponse.Projects
+
+	errGrp, downloadsCtx := errgroup.WithContext(context.Background())
+	for component, pkgs := range componentSpec {
+		for _, platform := range platforms {
+			targetPath := filepath.Join(dropPath)
+			err := os.MkdirAll(targetPath, 0755)
+			if err != nil {
+				return fmt.Errorf("failed to create directory %s", targetPath)
+			}
+			if mg.Verbose() {
+				log.Printf(">>>>>>> Prepare to download project [%s] for [%s]", component, platform)
+			}
+			for _, pkg := range pkgs {
+				reqPackage := platformPackages[platform]
+				pkgURL := resolveManifestPackage(projects[component], pkg, reqPackage, manifestResponse.Version)
+				if pkgURL != nil {
+					for _, p := range pkgURL {
+						log.Printf(">>>>>>>>> Downloading [%s] [%s] ", pkg, p)
+						pkgFilename := path.Base(p)
+						downloadTarget := filepath.Join(targetPath, pkgFilename)
+						if _, err := os.Stat(downloadTarget); err != nil {
+							func(downloadUrl string, target string) {
+								parsedURL, errorUrl := url.Parse(downloadUrl)
+								if errorUrl != nil {
+									errGrp.Go(func() error { return errorInvalidManifestURL })
+								}
+								var valid = false
+								for _, manifestHost := range AllowedManifestHosts {
+									if manifestHost == parsedURL.Host {
+										valid = true
+									}
+								}
+								if !valid {
+									log.Printf("Not allowed %s, valid ones are %+v", parsedURL.Host, AllowedManifestHosts)
+									errGrp.Go(func() error { return errorNotAllowedManifestURL })
+								}
+								cleanUrl := fmt.Sprintf("https://%s%s", parsedURL.Host, parsedURL.Path)
+								download := func() (string, error) { return downloadFile(downloadsCtx, cleanUrl, target) }
+								errGrp.Go(func() error { _, err := doWithRetries(download); return err })
+							}(p, downloadTarget)
```

Comment:
This does not look quite right: the function itself should be run in an `errGrp.Go()` statement not just the errors.
I will try to fix it and run the pipeline once for testing...